lib/directory_diff/transformer/temp_table.rb

```diff
@@ -0,0 +1,238 @@
+            # the current directory
+            new_records = csv_records.select("'insert'::varchar operation, row_number")
+                            .select(:name, :email, :phone_number, :assistants, :extra)
+                            .where({ employees.name => { email: nil } })
+            # deleted records are records in the current directory that don't
+            # exist in the new directory
+            deleted_records = employee_records.select("'delete'::varchar operation, row_number")
+                                .select(:name, :email, :phone_number, :assistants, :extra)
+                                .where({ csv.name => { email: nil } })
+            # changed records are records that have difference in name, phone
+            # number and/or assistants
+            changed_records = csv_records.select("'update'::varchar operation, row_number")
+                                .select(:name, :email, :phone_number, :assistants, :extra)
+                                .where.not(attributes_unchanged)
+            # unchanged records are records that are exactly the same in both
+            # directories (without considering the extra field)
+            unchanged_records = csv_records.select("'noop'::varchar operation, row_number")
+                                  .select(:name, :email, :phone_number, :assistants, :extra)
+                                  .where(attributes_unchanged)
+
+            # create temp table for holding operations
+            operations_temp_table = "temporary_operations_#{self.object_id}"
+            deduped_csv.connection.with_temporary_table operations_temp_table, new_records.to_sql do |name|
+              dec = ActiveRecordPgStuff::Relation::TemporaryTable::Decorator.new csv_records.klass, name
+              rel = ActiveRecord::Relation.new dec, table: dec.arel_table
+              rel.readonly!
+
+              rel.connection.execute("insert into #{name}(operation, row_number, name, email, phone_number, assistants, extra) #{deleted_records.to_sql}")
+              rel.connection.execute("insert into #{name}(operation, row_number, name, email, phone_number, assistants, extra) #{changed_records.to_sql}")
+
+              if options[:skip_noop] != true
+                rel.connection.execute("insert into #{name}(operation, row_number, name, email, phone_number, assistants, extra) #{unchanged_records.to_sql}")
+              end
+
+              rel.order(:row_number).each do |operation|
+                add_operation(operation)
+              end
+            end
+          end
+        end
+
+        prioritize_assistants(operations)
+      end
+
+      private
+
+      def add_operation(operation)
+        op = [
+          operation.operation.to_sym,
+          operation.name,
+          operation.email,
+          operation.phone_number.presence,
+          serialize_pg_array(operation.assistants)
+        ]
+        op << operation.extra unless operation[:extra].nil?
```

Comment:
@kamal what is this for?

Reply:
If the CSV has an extra column after assistant, we serialize it out. One of the tests show this behavior with the extra field `'foo'`